frontend/src/components/Login.js

```diff
@@ -0,0 +1,136 @@
+    event.preventDefault()
+    dispatch(login(name, email, password))
+    setName('')
+    setEmail('')
+    setPassword('')
+  }
+
+  const handleSignUp = (event) => {
+    event.preventDefault()
+    dispatch(signUp(name, email, password))
+    setName('')
+    setEmail('')
+    setPassword('')
+  }
+
+  return (
+    <Container>
+      <Wrapper>
+        {!accessToken ? (
+          <Form onSubmit={(event) => event.preventDefault()}>
+            <Title>Log in/Sign up</Title>
+            <SubTitle>to reveal your secret message...</SubTitle>
+            <TextField
+              required id="standard-required"  //adds *
+              label="Name"
+              inputProps={minimumNameLength}
+              value={name}
+              onChange={(event) => setName(event.target.value)}
+              helperText={name === "" ? 'min. 2 char' : ' '}
+            />
+            <div>
+              <InputLabel htmlFor="standard-adornment-password">Password*</InputLabel>
+              <Input
+                required={true}
+                id="standard-adornment-password"
+                type={showPassword ? "text" : "password"} //changed when icon is clicked
+                value={password}
+                fullWidth={true}
+                onChange={(event) => setPassword(event.target.value)}
+                inputProps={minimumPasswordLength}
+                endAdornment={
+                  <InputAdornment position="end">
+                    <IconButton
+                      aria-label="toggle password visibility"
+                      onClick={handleClickShowPassword}
+                    >
+                      {/* toggles */}
+                      {showPassword ? <Visibility /> : <VisibilityOff />}
+                    </IconButton>
+                  </InputAdornment>
+                }
+              />
+            </div>
+
+            <TextField
+              required id="standard-default"  //adds *
+              label="Email"
+              value={email}
+              onChange={(event) => setEmail(event.target.value)}
+              inputProps={validEmail}
+              helperText={email === "" ? 'x@xxx.xx' : ' '}
+            />
+
+            <ButtonWrapper>
+              <CustomButton
+                variant="contained"
+                color="default"
+                size="large"
+                type='submit'
+                onClick={handleSignUp}
+                text='Sign up'
+                disabled={!name || !password || !email}
+              />
+              <CustomButton
+                variant="contained"
+                color="primary"
+                size="large"
+                type='submit'
+                onClick={handleLogin}
+                text='Log in'
+                disabled={!name || !password || !email}
+              />
+            </ButtonWrapper>
+            {errorMessage && <SubTitle>{errorMessage}</SubTitle>}
+          </Form>
+        ) : (
+            <Profile />
```

Comment:
An alternative approach would be to have a page-component where you conditionally render login/signup-forms and profile.